include/mmtf/structure_data.hpp

```diff
@@ -63,10 +65,56 @@ struct GroupType { // NOTE: can't use MSGPACK_DEFINE_MAP due to char
         elementList == c.elementList &&
         bondAtomList == c.bondAtomList &&
         bondOrderList == c.bondOrderList &&
+        bondResonanceList == c.bondResonanceList &&
         groupName == c.groupName &&
         singleLetterCode == c.singleLetterCode &&
         chemCompType == c.chemCompType);
     }
+
+    std::string
+    as_string() const {
```

Comment:
what's the use-case of these `to_string()` methods? Debugging? How about implementing `operator<<` instead? The implementation could even go into a separate header (only declare a `friend` here) which is optional to include (e.g. `<mmtf/debug.hpp>`).

Reply:
Debugging, or general reporting.

I prefer to_string/as_string, that's really the only reason.  We could do move everything to a new file, but these functions don't really add much bloat. If you are passionate about the `<<` operator or the extra file I can switch to that, i just think it is a messier implementation.

Reply:
Thanks @danpf. No really strong feelings here, but my general take:
- I think these methods should have documentation which states their purpose
- I think adding these methods should be a separate pull request
- I think `operator<<` is cleaner

Reply:
I agree with @speleo3. The `operator<<` is the standard C++ way for something like this and anything else would be confusing. Plus the implementation is almost identical since you are using streams anyways. If the string output looks decent, it can stay in the class as the standard string representation. Otherwise, I would also vote for having it in a separate header.